src/main/core/work/task.rs

```diff
@@ -0,0 +1,140 @@
+use std::sync::Arc;
+
+use atomic_refcell::AtomicRefCell;
+
+use crate::{core::worker::Worker, host::host::Host};
+
+/// Mostly for interoperability with C APIs.
+/// In Rust code that doesn't need to interact with C, it may make more sense
+/// to directly use a `FnMut(&mut Host)` trait object.
+pub struct Task {
+    inner: Arc<AtomicRefCell<dyn FnMut(&mut Host)>>,
+    #[cfg(debug_assertions)]
+    magic: u32,
+}
```

Comment:
Good catch. Added the `IsSend` and `IsSync` traits as well.

Could use a second pair of eyes on e.g. the `SAFETY` comments where we deal with raw pointers. I think `SendSyncPointer` perhaps ought to similarly make its member private, and its methods `unsafe` with appropriate `SAFETY` comments, but opted to skip that rabbit hole for now :).

Reply:
I think `SyncSendPointer` is fine here since that's what the current C code does here. My only concern is that `Sync` and `Send` give us memory safety, but don't provide the correctness guarantees that we want (that a closure containing a shared reference belonging to a host is only accessed when that host's lock is held). I made a comment below where I tried to explain my thoughts.

> I think `SendSyncPointer` perhaps ought to similarly make its member private, and its methods `unsafe` with appropriate `SAFETY` comments, but opted to skip that rabbit hole for now :)

I wouldn't be against that, but maybe only make a `SyncSendPointer::new()` constructor unsafe. I don't think methods like `ptr()` need to be unsafe since they don't dereference or do anything that's unsafe.

Reply:
>  My only concern is that ...

Addressed (to the extent feasible) with additional comments and the Host runtime check.

> I wouldn't be against that, but maybe only make a SyncSendPointer::new() constructor unsafe. I don't think methods like ptr() need to be unsafe since they don't dereference or do anything that's unsafe.

Maybe; we can cross that bridge when we come to it :). e.g. I had started writing a `SAFETY` comment for `SyncSendPointer::new` and found myself writing about conditions that must hold at the time that the pointer  is accessed, which seems more appropriate to put on the accessors.